robotoff/insights/annotate.py

```diff
@@ -474,6 +475,61 @@ def process_annotation(
         return UPDATED_ANNOTATION_RESULT
 
 
+class UPCImageAnnotator(InsightAnnotator):
+    """Annotation for UPC images
+    Unselect the product images that match a UPC_Image
+    A UPC_Image is defined as an image that has a UPC with a high pct area in the image and thus it is a poor quality main photo
+    """
+
+    @classmethod
+    def process_annotation(
+        cls,
+        insight: ProductInsight,
+        data: Optional[dict] = None,
+        auth: Optional[OFFAuthentication] = None,
+        is_vote: bool = False,
+    ) -> AnnotationResult:
+        product_id = insight.get_product_id()
+        product = get_product(product_id, ["images"])
+
+        image_id = get_image_id(insight.source_image or "")
+
+        if product is None:
+            logger.info(
+                "Product %s not found, cannot unselect image %s", product_id, image_id
+            )
+            return MISSING_PRODUCT_RESULT
+
+        if image_id is None:
+            logger.info("Image %s not found, cannot unselect", image_id)
+            return FAILED_UPDATE_RESULT
+
+        to_unselect = []
+
+        images = product["images"]
+        if image_id not in images:
+            logger.info("Image %s not found in %s", image_id, product_id)
+            return FAILED_UPDATE_RESULT
+
+        for image_field, image_data in (
+            (key, data) for key, data in images.items() if not key.isdigit()
+        ):
+            if image_data["imgid"] == image_id:
+                to_unselect.append(image_field)
+
+        for image_field in to_unselect:
+            logger.info(
+                "Sending unselect request for image %s of %s", image_field, product_id
+            )
+            if not auth:
+                logger.info("No auth provided, cannot unselect image")
+                return FAILED_UPDATE_RESULT
+
```

Comment:
If there are no auth robotoff default credentials are used, we shoud support this.
```suggestion
```

edit: we should update `unselect_image` in `off.py` to support non-user authenticated writes (=writes in the name of robotoff user) with a `params.update(off_credentials())`

Reply:
Done!